python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -56,14 +57,49 @@
 from dagster._utils.merger import merge_dicts
 from dagster._utils.warnings import deprecation_warning
 
-from .compute import OpExecutionContext
+from .compute import AssetExecutionContext, OpExecutionContext
 from .system import StepExecutionContext, TypeCheckContext
 
 
 def _property_msg(prop_name: str, method_name: str) -> str:
-    return (
-        f"The {prop_name} {method_name} is not set on the context when an op is directly invoked."
-    )
+    return f"The {prop_name} {method_name} is not set on the context when an asset or op is directly invoked."
+
+
+class BaseDirectExecutionContext:
```

Comment:
doc string why this exists so its clear to future refactor-er when it would be ok to change the set-up